src/Backups/BackupEntryWrappedWith.h

```diff
@@ -0,0 +1,37 @@
+#pragma once
+
+#include <Backups/IBackupEntry.h>
+
+
+namespace DB
+{
+
+/// Wraps another backup entry and a value of any type.
+template <typename T>
+class BackupEntryWrappedWith : public IBackupEntry
```

Comment:
It's easy to forget to add new method here after updating `IBackupEntry` interface, but ok.

Reply:
Maybe we can move this class to `IBackupEntry.h` to make it harder to forget

Reply:
> It's easy to forget to add new method here after updating `IBackupEntry` interface, but ok.
I'd prefer to keep each class in a separate header. We won't forget to override pure virtual methods anyway.

Reply:
But not all methods are pure virtual, for example, `getChecksum()` has default implementation in the base class.